File: Java/min-heap.java
import java.util.PriorityQueue;

public class Main {
    public static void main(String[] args) {
        PriorityQueue<Integer> heap = new PriorityQueue<>();
        heap.add(5);
        heap.add(2);
        heap.add(8);
        
        System.out.println(heap.poll());  // 2 (минимальный элемент)
    }
}

File: Python/min-heap.cpp
#include <iostream>
#include <queue>
#include <vector>

int main() {
    std::priority_queue<int, std::vector<int>, std::greater<int>> heap;
    heap.push(5);
    heap.push(2);
    heap.push(8);
    
    std::cout << heap.top() << std::endl;  // 2
    heap.pop();
    
    return 0;
}
